src/infrastructure/channels/reliable_channel.rs

```diff
@@ -141,9 +141,19 @@ impl Channel for ReliableChannel {
 
         self.dropped_packets = dropped_packets.into_iter().map(|(_, p)| p).collect();
 
-        let a = buffer.len();
-        let b = acked_header.size();
-        // TODO: resent packets if there are dropped packets
         Ok(&buffer[acked_header.size() as usize .. buffer.len()])
     }
+}
+
+impl ReliableChannel {
+    /// Check if this channel has dropped packets.
+    /// Performance wise it is better to use this to check if there are dropped packets than calling 'get_dropped_packets directly()'.
+    pub fn has_dropped_packets(&self) -> bool {
+        !self.dropped_packets.len() == 0
```

Comment:
please use `is_empty` here

Reply:
Please make sure to explain why when making these comments.

Reply:
Oké, I'll have some more detailed comment. Also why use 'is empty', question I than have is what is empty? I'll see what I can do for a better comment.

Reply:
`len() == 0` is a common enough pattern I don't see it as unclear. Also not all things implement is_empty().

Reply:
Its about reducing thought when its comes to reading a line having a `!somecode == 0` takes a lot more thought than `!somecode.is_empty()`.

Reply:
Why do you think it takes a lot more thought?

(Yes, I am challenging a base assumption)

Reply:
because `is_empty` is explicit, its reads like english. `len() == 0` takes slightly more time to think about when reading it I find.I don't see why you would pick `len() == 0` over `is_empty`?

Reply:
`len() == 0` is just as explicit. You're right in that is_empty() is much English-ish. Should that be a primary concern when many of our devs speak English as a 2nd language?

Reply:
I mean, the codebase is in english so therefore it should try and flow as English as possible?

Reply:
Excellence does not require perfection.
Perfection is the enemy of progress.

Reply:
I have implemented `is_empty()` instead of manually checking the length of the vector. I did not know of the function `is_empty()` but it basically does the same as you look at stdlib.